### DISCUSSION TITLE
Why does Mongock need MongoDb with ReplicaSet? Can it work on MongoDB without need of setting up ReplicaSet?

### DISCUSSION THREAD

**Discussion by Rachit-zymr** [flamingock/mongock#17] [Q&A] [answered]
In the [Requirements ](https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-advance#requirements) section provided in Examples, one of the requirements says: "MongoDB with ReplicaSet",
We would like to know the following:
Why does Mongock library need MongoDB with ReplicaSet?
Will Mongock work well with just MongoDB (without a ReplicaSet)?
Ref:
https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-advance#requirements

Apart from mentioning in the Requirements section, is the above requirement mentioned anywhere in the [official documentation](https://docs.mongock.io/) with a proper explanation?

Reason behind asking the question: We are planning to use Mongock for MongoDB Database Migration, we may not always want to use ReplicaSets or replication for MongoDB

**Comment by dieppa** [comment, marked as answer]
Hello @Rachit-zymr ,

first, thanks for using Mongock.

 
Let me explain a couple of things first

- That [document](https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-advance#requirements) you mention is only referring to that example. It's not applicable to Mongock in general. For official documentation, please go to [this link](https://docs.mongock.io/)

- Mongock doesn't force you to use replicaSets. Mongock is happy as long as MongoDB is happy 😄

In this case that example uses MongoDB transactions and that requires replicaSets. But that requirement doesn't come from Mongock, but from MongoDB itself.

I hope you find this helpful.
If you have any other question, please don't hesitate to raise a discussion or issue.

Thanks

**Reply by RoniBam** [reply]
Hi, can you please mention where is the use of the mongoDB transaction? is it related to the secondary db use?

**Reply by dieppa** [reply]
The transaction is required to make sure the user changes(the actual changes  you do in the changeUnit) and the changelog is saved atomically. It's not mandatory, but without that, Mongock would need to run the rollback and it's not guaranteed that it finish successfully
